src/Controllers/BatchProcessorController.ts

```diff
@@ -89,100 +103,312 @@ module StreamStats.Controllers {
+
+            regressionRegions.forEach((regressionRegion) => {
+
+                regressionRegion.parameters.forEach((parameter) => {
+                    var paramCode = parameter.code;
+                    for (var i = 0; i < this.availableParamList.length; i++) {
+                        let p = this.availableParamList[i];
+
+                        if (p.code.toUpperCase() === paramCode.toUpperCase()) {
+                            p['checked'] = true;
+                            p['toggleable'] = false;
+                            break;
+                        }
+                    }
+                });
             });
+        }
 
-            // toggle switch
-            this.flowStatsAllChecked = !this.flowStatsAllChecked;
+
+        public onSelectedStatisticsGroupChanged(): void {
+
+            //loop over whole statisticsgroups
+            this.selectedFlowStatsList.forEach((statisticsGroup) => {
+
+                // set checked to true
+                statisticsGroup.checked = true;
+
+                if (statisticsGroup.regressionRegions) {
+
+                    //get their parameters
+                    statisticsGroup.regressionRegions.forEach((regressionRegion) => {
+
+                        //loop over list of state/region parameters to see if there is a match
+                        regressionRegion.parameters.forEach((param) => {
+
+                            var found = false;
+                            for (var i = 0; i < this.availableParamList.length; i++) {
+                                var parameter = this.availableParamList[i];
+                                if (parameter.code.toLowerCase() == param.code.toLowerCase()) {
+                                    this.addParameterToSelectedParamList(param.code);
+                                    found = true;
+                                    break;
+                                }//end if
+                            }//next iparam
+
+                            if (!found) {
+                                // this.toaster.pop('warning', "Missing Parameter: " + param.code, "The selected scenario requires a parameter not available in this State/Region.  The value for this parameter will need to be entered manually.", 0);
+
+                                //add to region parameterList
+                                var newParam = {
+                                    name: param.name,
+                                    description: param.description,
+                                    code: param.code,
+                                    unit: param.unitType.unit,
+                                    value: null,
+                                    regulatedValue: null,
+                                    unRegulatedValue: null,
+                                    loaded: null,
+                                    checked: false,
+                                    toggleable: true

```

Comment:
We don't need the additional properties
```suggestion
                                var newParam = {
                                    code: param.code,
                                    description: param.description,
                                    checked: false,
                                    toggleable: true

```